tests/integration/integration-test.py

```diff
@@ -79,93 +85,166 @@ def test_get_notes_query_by_student_id(self, endpoint='/notes'):
+            greater_element = datetime.strptime(last_modified,
+                                                '%Y-%m-%dT%H:%M:%S.%f%z')
+            for resource in response:
+                current_element = resource['attributes']['lastModified']
+                current_element = datetime.strptime(current_element,
+                                                    '%Y-%m-%dT%H:%M:%S.%f%z')
+                self.assertGreaterEqual(greater_element, current_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
+    def check_alphabetical_sort(self, response, sort_key):
+        try:
+            greater_element = response[0]['attributes'][sort_key]
+            for resource in response:
+                current_element = resource['attributes'][sort_key]
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
+    def check_context_type_sort(self, response):
+        try:
+            attributes = response[0]['attributes']
+            greater_element = attributes['context']['contextType']
+            for resource in response:
+                attributes = resource['attributes']
+                current_element = attributes['context']['contextType']
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        except IndexError:
+            self.assertEqual(response, [])
+
     def query_source(self, student_id, sources, endpoint='/notes'):
         for source in sources:
             params = {'studentId': student_id, 'sources': source}
-            response = utils.make_request(self, endpoint, 200, params=params)
-            schema = utils.get_resource_schema(self, 'NoteResource')
-            utils.check_schema(self, response, schema)
-            # Validating the source requested is the same source 
+            response = self.get_response(params)
+            # Validating the source requested is the same source
             for resource in response.json()['data']:
                 actual_source = resource['attributes']['source']
                 self.assertEqual(actual_source, source)
-
-    def query_sort_keys(self, student_id, endpoint='/notes'):
-        sort_keys = ['lastModified', 'source', 'permissions'] # , 'contextType']
-        for sort_key in sort_keys:
-            params = {'studentId': student_id, 'sortKey': sort_key}
-            response = utils.make_request(self, endpoint, 200, params=params)
-            note_schema = utils.get_resource_schema(self, 'NoteResource')
-            utils.check_schema(self, response, note_schema)
+        # Check more than one source at a time
+        if len(sources) > 0:
+            array_len = random.randint(1, len(sources))
```

Comment:
It doesn't seem like this `array_len` is array length. I'd recommend renaming it more descriptive, e.g. `sample_numbers`